server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -48,85 +48,90 @@ public static MappingStats of(Metadata metadata, Runnable ensureNotCancelled) {
         Map<String, FieldStats> fieldTypes = new HashMap<>();
         Set<String> concreteFieldNames = new HashSet<>();
         Map<String, RuntimeFieldStats> runtimeFieldTypes = new HashMap<>();
+        final Map<String, Integer> mappingCounts = Maps.newMapWithExpectedSize(metadata.getMappingsByHash().size());
         for (IndexMetadata indexMetadata : metadata) {
-            ensureNotCancelled.run();
```

Comment:
I think we can keep calling `ensureNotCancelled.run()` here too?

Reply:
It seems really trivial here. This is always going to run in no-time compared to the rest of the method (it's just a simple map put here compared to deserialization in the other loop) no matter the number of indices in the cluster => didn't think it was really worth it adding it to this loop, if we add it here then we could make a similar argument to add it to a bunch of other computationally equally cheap places which felt wrong to me?